src/tests/ftest/util/dmg_utils.py

```diff
@@ -44,72 +44,266 @@ def __init__(self, path):
+            super(DmgCommand.StorageSubCommand, self).__init__(
+                "/run/dmg/storage/*", "storage")
+
+        def get_sub_command_class(self):
+            """Get the dmg storage sub command object."""
+            if self.sub_command.value == "format":
+                self.sub_command_class = self.FormatSubCommand()
+            elif self.sub_command.value == "prepare":
+                self.sub_command_class = self.PrepareSubCommand()
+            elif self.sub_command.value == "query":
+                self.sub_command_class = self.QuerySubCommand()
+            elif self.sub_command.value == "scan":
+                self.sub_command_class = self.ScanSubCommand()
+            else:
+                self.sub_command_class = None
+
+        class FormatSubCommand(CommandWithParameters):
+            """Defines an object for the dmg storage format command."""
+
+            def __init__(self):
+                """Create a dmg storage format command object."""
+                super(
+                    DmgCommand.StorageSubCommand.FormatSubCommand,
+                    self).__init__(
+                        "/run/dmg/storage/format/*", "format")
+                self.reformat = FormattedParameter("--reformat", False)
+
+        class PrepareSubCommand(CommandWithParameters):
+            """Defines an object for the dmg storage format command."""
+
+            def __init__(self):
+                """Create a dmg storage prepare command object."""
+                super(
+                    DmgCommand.StorageSubCommand.PrepareSubCommand,
+                    self).__init__(
+                        "/run/dmg/storage/prepare/*", "prepare")
+                self.pci_whitelist = FormattedParameter("-w {}", None)
+                self.hugepages = FormattedParameter("-p {}", None)
+                self.target_user = FormattedParameter("-u {}", None)
+                self.nvme_only = FormattedParameter("-n", False)
+                self.scm_only = FormattedParameter("-s", False)
+                self.reset = FormattedParameter("--reset", False)
+                self.force = FormattedParameter("-f", False)
+
+        class QuerySubCommand(CommandWithSubCommand):
+            """Defines an object for the dmg query format command."""
+
+            def __init__(self):
+                """Create a dmg storage query command object."""
+                super(
+                    DmgCommand.StorageSubCommand.QuerySubCommand,
+                    self).__init__(
+                        "/run/dmg/storage/query/*", "query")
+
+            def get_sub_command_class(self):
+                """Get the dmg pool sub command object."""
+                if self.sub_command.value == "blobstore-health":
+                    self.sub_command_class = self.BlobstoreHealthSubCommand()
+                elif self.sub_command.value == "smd":
+                    self.sub_command_class = self.SmdSubCommand()
```

Comment:
(pylint-redefined-variable-type) Redefinition of self.sub_command_class type from util.dmg_utils.DmgCommand.StorageSubCommand.QuerySubCommand.BlobstoreHealthSubCommand to util.dmg_utils.DmgCommand.StorageSubCommand.QuerySubCommand.SmdSubCommand